src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -1607,5 +1607,145 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> other,
+            if (array!.Length - index < length) // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_InvalidOffLen);
+
+            if (length > 1)
+            {
+#if CORECLR
+                if (comparer == null || comparer == Comparer<T>.Default)
+                {
+                    // Array based
+                    //if (TrySZSort(array, null, index, index + length - 1))
+                    //{
+                    //    return;
+                    //}
+                    unsafe
+                    {
+                        ref byte byteRef = ref Unsafe.As<T, byte>(ref array.GetPinnableReference());
+                        fixed (byte* ptr = &byteRef)
+                        {
+                            if (Array.TrySZSort(new IntPtr(ptr), typeof(T), IntPtr.Zero, typeof(T), index, index + length - 1))
+                            {
+                                return;
+                            }
+                        }
+                    }
+                }
+#endif
+
+                ArraySortHelper<T>.Default.Sort(array, index, length, comparer);
+            }
+        }
+
+        public static void Sort<TKey, TValue>(this Span<TKey> keys, Span<TValue> items, int index, int length, System.Collections.Generic.IComparer<TKey>? comparer)
+        {
+            if (keys == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.keys);
+            if (index < 0)
+                ThrowHelper.ThrowIndexArgumentOutOfRange_NeedNonNegNumException();
+            if (length < 0)
+                ThrowHelper.ThrowLengthArgumentOutOfRange_ArgumentOutOfRange_NeedNonNegNum();
+            if (keys!.Length - index < length || (items != null && index > items.Length - length)) // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_InvalidOffLen);
+
+            if (length > 1)
+            {
+#if CORECLR
+                if (comparer == null || comparer == Comparer<TKey>.Default)
+                {
+                    // Array based
+                    //if (TrySZSort(keys, items, index, index + length - 1))
+                    //{
+                    //    return;
+                    //}
+                    unsafe
+                    {
+                        ref byte keysRef = ref Unsafe.As<TKey, byte>(ref keys.GetPinnableReference());
+                        ref byte itemsRef = ref Unsafe.As<TValue, byte>(ref items.GetPinnableReference());
+                        fixed (byte* keysPtr = &keysRef)
+                        fixed (byte* itemsPtr = &itemsRef)
+                        {
+                            if (Array.TrySZSort(new IntPtr(keysPtr), typeof(TKey), new IntPtr(itemsPtr), typeof(TValue), index, index + length - 1))
```

Comment:
Need a way to either call `TrySZSort` here for spans as suggested here, or spans can revert to slower managed code.